consensus/ibft/fork/hooks.go

```diff
@@ -0,0 +1,91 @@
+package fork
+
+import (
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/consensus/ibft/hook"
+	"github.com/0xPolygon/polygon-edge/contracts/staking"
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+	stakingHelper "github.com/0xPolygon/polygon-edge/helper/staking"
+	"github.com/0xPolygon/polygon-edge/state"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/0xPolygon/polygon-edge/validators"
+	"github.com/0xPolygon/polygon-edge/validators/valset"
+)
+
+func registerPoSHook(
+	hooks *hook.HookManager,
+	epochSize uint64,
+) {
+	isLastEpoch := func(height uint64) bool {
+		return height > 0 && height%epochSize == 0
+	}
+
+	hooks.ShouldWriteTransactionFunc = func(height uint64) bool {
+		return !isLastEpoch(height)
+	}
+
+	hooks.VerifyBlockFunc = func(block *types.Block) error {
+		if isLastEpoch(block.Number()) && len(block.Transactions) > 0 {
+			return fmt.Errorf("block can't have transactions in the last of epoch")
```

Comment:
Shouldn't be an `Errorf`

Reply:
Defined error as constant at 11497ad01edd81dd515f826b479650217be68ff5